otter/log/formatters.py

```diff
@@ -215,6 +219,42 @@ def audit_log_formatter(eventDict, timestamp, hostname):
     return audit_log_params
 
 
+def ErrorFormattingWrapper(observer):
+    """
+    Return log observer that will format error if any and delegate it to
+    given `observer`
+    """
+
+    def error_formatting_observer(event):
+
+        message = ""
+
+        if event.get("isError", False):
+            level = 3
+
+            if 'failure' in event:
+                excp = event['failure'].value
+                message = repr(excp)
+                event['traceback'] = event['failure'].getTraceback()
+                event['exception_type'] = excp.__class__.__name__
+
+            if 'why' in event and event['why']:
+                message = '{0}: {1}'.format(event['why'], message)
+
+        else:
+            level = 6
+
+        event.update({
+            "message": event.get("message", (message, )),
+            "level": event.get("level", level)
+        })
+        event.pop('isError', None)
```

Comment:
Do we want to remove all the `ERROR_FIELDS`?  Since the useful failure information is being translated here, for instance, `failure` may no longer be necessary.